feat: add readme for the migration tool (#4144)
ignite/cli: ignite/internal/tools/gen-mig-diffs/readme.md

<div align="center">
  <h1> Generate Ignite Migration Diffs </h1>
</div>

This repository hosts the Chain Scaffold Migration Tool for Ignite CLI, designed to help developers migrate their

> projects from older versions to the latest release of the Ignite CLI framework.

Reviewer: ```suggestion
projects from older versions of Ignite to the latest release.
```

This tool addresses compatibility and feature alignment as a detailed
in [Issue #3699](https://github.com/ignite/cli/issues/3699) and implemented
in [PR #3718](https://github.com/ignite/cli/pull/3718).

> The migration tool aims to streamline the update process for projects built with Ignite CLI, ensuring they leverage the

Reviewer: If moving the feature list here, we can omit this as it is covered in the list

latest improvements and SDK stack.

> ## Quick start

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

> ### Installation

Reviewer: ```suggestion
## Installation
```

> This tool is part of the internal tools of the Ignite CLI, located in the `ignite/internal/tools/gen-mig-diffs`

Reviewer: ```suggestion
It is located in the `ignite/internal/tools/gen-mig-diffs`
```

directory and made it a standalone project.

To set up this tool in your development environment:

> - Clone the Ignite CLI repository:

Reviewer: ```suggestion
1. Clone the Ignite CLI repository:
```

Reviewer: Numbered list would be preferred so they are presented as sequential steps. We should also indent the codeblocks so that they align with each point.

```shell
git clone https://github.com/ignite/cli.git && \
cd cli/ignite/internal/tools/gen-mig-diffs
```

> - Install and show usage:

Reviewer: ```suggestion
2. Install and show usage:
```

```shell
go install . && gen-mig-diffs -h
```

> - Run migration diff tools:

Reviewer: ```suggestion
3. Run migration diff tool:
```

```shell
gen-mig-diffs --output temp/migs --from v0.27.2 --to v28.3.0
```

## Usage

```bash
This tool is used to generate migration diff files for each of ignites scaffold commands

Usage:
  gen-mig-diffs [flags]

Flags:
  -f, --from string              Version of Ignite or path to Ignite source code to generate the diff from
  -h, --help                     help for gen-mig-diffs
  -o, --output string            Output directory to save the migration document (default "docs/docs/06-migration")
      --repo-output string       Output path to clone the Ignite repository
  -s, --repo-source string       Path to Ignite source code repository. Set the source automatically set the cleanup to false
      --repo-url string          Git URL for the Ignite repository (default "https://github.com/ignite/cli.git")
      --scaffold-cache string    Path to cache directory
      --scaffold-output string   Output path to clone the Ignite repository
  -t, --to string                Version of Ignite or path to Ignite source code to generate the diff to
```

> ## Features

Reviewer: We could move this section before installation

Reviewer: It could also just be the points stating the features as part of the initial explanation
